### ISSUE TITLE
Floating element flickers / flips continuously on overflow due to appearing scrollbars

### ISSUE THREAD

**Issue by m3l1x** [floating-ui/floating-ui#2822]
**Describe the bug** A clear and concise description of what the bug is.
Floating element with flip middleware flickers while resizing the window / flips continuously to the opposite site for the width of the appearing scrollbar when it overflows the viewport. By overflowing of the right boundary a vertical scrollbar is being automatically triggered due to width: 100vw and height: 100vh on any element.

**To Reproduce**
https://codesandbox.io/p/sandbox/ecstatic-kowalevski-n5wrdc?file=%2Fsrc%2Findex.js

1. Go to the provided sandbox
2. Slowly move the editor sash to reduce embedded window size
3. As soon as the floating element begins to overflow, it starts to flip from right to left and vice versa until the appearing/disappearing scrollbar does not affect floating-uis logic for "bestFit"
4. When you stop resizing in the described area, it happens that the flipping continues infinitely.

**Expected behavior** A clear and concise description of what you expected to
happen.
Floating element should not flip, if it would trigger an overflow

**Screenshots** If applicable, add screenshots to help explain your problem.
![screenshot_floating](https://github.com/floating-ui/floating-ui/assets/33684047/b9edc855-ae4f-4a2f-b01b-276de886ac0b)

**Context:**

- OS: Windows
- Browser: Chrome 123.0.6312.59, Firefox 124.0

**Comment by atomiks** [comment]
> triggered due to width: 100vw and height: 100vh on any element.

I don't think this particular issue can be fixed easily. I recommend using the `size` middleware to prevent it from overflowing in the first place.
